### PR TITLE
SALTO-1482 Workflow Instance are represented in the NaCL but not in Salesforce UI

### PR DESCRIPTION
**Pull request by michalshavit1**
removed the Workflow Instance are representation in the NaCL 

---

_Release Notes_: 
Workflow Instance will be no longer represented in the NaCL

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 1]
```diff
@@ -194,8 +194,10 @@ const filterCreator: FilterCreator = () => {
           }
           const innerInstances = await Promise.all(makeArray(workflowInst.value[fieldName])
             .map(async innerValue => {
-              innerValue[INSTANCE_FULL_NAME_FIELD] = fullApiName(await apiName(workflowInst),
-                innerValue[INSTANCE_FULL_NAME_FIELD])
+              innerValue[INSTANCE_FULL_NAME_FIELD] = fullApiName(
+                await apiName(workflowInst),
+                innerValue[INSTANCE_FULL_NAME_FIELD]
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/filters/workflow.ts:199]: Super super nit: The reason why I prefer the ',' anyway is for cleaner commits in the future. If I were to add/remove any parameter from this function I would only see in the diff the line which changed and not the paremeter before as well.
```suggestion
                innerValue[INSTANCE_FULL_NAME_FIELD],
```

**Review by amitgoldarb — APPROVED** [review]
Great work!
Looks good to me, only 1 super nit.
@ori-moisis would you like to take a look as well?

File: packages/salesforce-adapter/e2e_test/workflow.test.ts [hunk 2]
```diff
@@ -188,32 +174,7 @@ describe('workflow filter', () => {
...
-      it('should fetch workflow task', async () => {
-        verifySubInstance(WORKFLOW_TASKS_FIELD, `${baseCustomObject}_TestWorkflowTask@v`,
-          'E2E Fetch WorkflowTask')
-      })
-    })
-    describe('fetch workflow rule', () => {
-      it('should fetch workflow rule', async () => {
-        verifySubInstance(WORKFLOW_RULES_FIELD, `${baseCustomObject}_TestWorkflowRule@v`,
-          'E2E Fetch WorkflowRule')
+        expect(workflow).toBeUndefined()
```
Comment (ori-moisis) [inline packages/salesforce-adapter/e2e_test/workflow.test.ts:177]: test description says the opposite of what the test checks

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 3]
```diff
@@ -194,8 +194,10 @@ const filterCreator: FilterCreator = () => {
...
+              )
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/workflow.ts:200]: can't comment on the relevant line, but in line 204 it seems like we are still editing the original workflow instance (the one we are about the remove)
why are we still doing that?

File: packages/salesforce-adapter/e2e_test/workflow.test.ts [hunk 4]
```diff
@@ -188,32 +174,7 @@ describe('workflow filter', () => {
     })
     describe('fetch workflow', () => {
       it('should fetch workflow', async () => {
-        expect(workflow.value[INSTANCE_FULL_NAME_FIELD]).toBe(baseCustomObject)
-      })
-    })
-    describe('fetch workflow alerts', () => {
```
Comment (ori-moisis) [inline packages/salesforce-adapter/e2e_test/workflow.test.ts:194]: seems like you removed all the test for the things the filter is doing with the nested instances.
we should not remove all the tests...

**Review by ori-moisis — COMMENTED** [review]
Can you please add an example of how the new instances look in the PR description?
I don't see the code that would put the object type as a parent, not sure if I am missing something or if it is really not implemented as required

**Review by ori-moisis — COMMENTED** [review]
@michalshavit1  I think we also saw things that needed to be removed from the `custom_object` filter, right?

File: packages/salesforce-adapter/e2e_test/workflow.test.ts [hunk 5]
```diff
@@ -164,11 +161,12 @@ describe('workflow filter', () => {
...
     }
 
-    const verifySubInstance = (subField: string, subName: string,
-      subDescription: string): void => {
-      expect(workflow.value[subField]).toBeDefined()
-      const subElemId = makeArray(workflow.value[subField])
-        .find((ref: {elemID: ElemID}) => ref.elemID.name === subName)?.elemID
+    const verifySubInstance = (
+      subFieldName: string,
+      subName: readonly string[],
```
Comment (ori-moisis) [inline packages/salesforce-adapter/e2e_test/workflow.test.ts:166]: ummm, why is this a list? seems like everywhere this is used it is only one string...

File: packages/salesforce-adapter/e2e_test/workflow.test.ts [hunk 6]
```diff
@@ -164,11 +161,12 @@ describe('workflow filter', () => {
       return verifyHasWorkflowRule() // WorkflowRule depends on Alert, FieldUpdate & Task
     }
 
-    const verifySubInstance = (subField: string, subName: string,
-      subDescription: string): void => {
-      expect(workflow.value[subField]).toBeDefined()
-      const subElemId = makeArray(workflow.value[subField])
-        .find((ref: {elemID: ElemID}) => ref.elemID.name === subName)?.elemID
+    const verifySubInstance = (
+      subFieldName: string,
```
Comment (ori-moisis) [inline packages/salesforce-adapter/e2e_test/workflow.test.ts:165]: this is not the field name, it is the type name

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 7]
```diff
@@ -137,7 +137,7 @@ const createDummyWorkflowInstance = async (
       metadataType: 'Workflow',
       dirName: 'workflows',
       suffix: 'workflow',
-    } as MetadataTypeAnnotations,
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/workflow.ts:140]: this cast was there on purpose (to ensure the annotations have the correct names and types), why did you remove it?

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 8]
```diff
@@ -149,20 +149,17 @@ const createDummyWorkflowInstance = async (
...
   // we assume the only possible changes are in nested instances and we omit the nested instance
   // fields from the parent when we create the partial instance. so we can use the same parent as
   // the after instance here
   const before = await createPartialWorkflowInstance(parent, changes, 'before')
-
+  /*
+   * we won't be able to know if the change was add or modified
+   * it does not matter in this use case since changes
+   * will go to salesforce upsert or add API anyway.
+   */
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/workflow.ts:162]: `upsert` is already `update or insert`, doesn't make sense to say `upsert or add`

```suggestion
  // we cannot know if the workflow instance change is add or modify
  // but it does not matter in this use case because changes
  // will be deployed with the upsert API anyway
```

also, now that we always use the dummy instance as a parent, the comment in lines 154-156 is not needed anymore and can be removed

**Review by ori-moisis — APPROVED** [review]
